src/test/jtx/envconfig.h

```diff
@@ -0,0 +1,105 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+#define RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+
+#include <ripple/core/Config.h>
+
+namespace ripple {
+namespace test {
+
+/// @brief initializes a config object for use with jtx::Env
+///
+/// @param config the configuration object to be initialized
+extern
+void
+setupConfigForUnitTests (Config& config);
+
+namespace jtx {
+
+/// @brief creates and initializes a default
+/// configuration for jtx::Env
+///
+/// @return unique_ptr to Config instance
+inline
+std::unique_ptr<Config>
+envconfig()
+{
+    auto p = std::make_unique<Config>();
+    setupConfigForUnitTests(*p);
+    return p;
```

Comment:
How about changing `modfunc` so it takes the `unique_ptr` as a parameter and returns a `unique_ptr`? (we could also rename it to `f`). So the whole function would be: `return f(envconf(), std::forward<Args>(args)...);`